pkg/loopback/loopback_test.go

```diff
@@ -0,0 +1,118 @@
+// +build linux,cgo
+
+package loopback
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+const filePath = "/var/tmp/"
+
+var rootEnabled bool
+
+func init() {
+	flag.BoolVar(&rootEnabled, "test.root", false, "enable tests that require root")
+}
+
+func RequiresRoot(t *testing.T) {
+	if !rootEnabled {
+		t.Skip("skipping test that requires root")
+		return
+	}
+	require.Equal(t, 0, os.Getuid(), "This test must be run as root.")
+}
+
+func createSparseFile() (string, error) {
+	size := int64(10 * 1024 * 1024)
+	fd, err := ioutil.TempFile(filePath, "sparseFile")
```

Comment:
We need to provide a filepath which is the mounted volume inside the docker container, otherwise the function `file.stat()` provides different values for device and inode. To bypass this we had to mount a folder and create a file in that so that we can get accurate system level values of device and inode.